#5474: Replace jetty with tomcat7 mvn plugin (#5719)
geosolutions-it/MapStore2: docs/developer-guide/developing.md

After a while (depending on the network bandwidth) the full set of dependencies and tools will be downloaded to the **node_modules** sub-folder.

* start the development instance with:

`npm start`

> Then point your preferred browser to [http://localhost:8081](http://localhost:8081). By default the front-end works using the online dev server as back-end. This configuration is useful on start-up quickly but is not a suggested configuration if you want to develop.

Reviewer: ```suggestion
Then point your preferred browser to [http://localhost:8081](http://localhost:8081). By default the front-end works using the online dev server as back-end. This configuration is useful for a quick startup, but is not the suggested configuration if you want to develop.
```

To learn how to connect the front-end dev server to a local back-end read the following instructions.

### Connect Front-end to local back-end

> By default `npm start` runs a dev server connected to the mapstore online dev server.

Reviewer: ```suggestion
By default `npm start` uses the online dev server as a backend.
```

> This configuration have to be changed to develop locally in order to access all the functionalities.

Reviewer: ```suggestion
This configuration needs to be changed to develop locally in order to access all the functionalities.
```

To use a local back-end you have to:

* **Remove auth configuration dedicated to GeoServer** from `localConfig.json --> authenticationRules` (it provides the MapStore/GeoServer integration for dev-server, that is not present in your local back-end)

```diff
"authenticationRules": [{
        "urlPattern": ".*geostore.*",
        "method": "bearer"
      }, {
        "urlPattern": ".*rest/config.*",
        "method": "bearer"
-      },
-      {
-        "urlPattern": "http(s)?\\:\\/\\/gs-stable\\.geo-solutions\\.it\\/geoserver/.*",
-        "authkeyParamName": "authkey",
-       "method": "authkey"
    }],
```

* **Run the back-end locally**. See the [dedicated section in this page](#Back-end)

> * **Setup dev-server to use the local back-end**, applying this changes to `buildConfig.js` --> devServer configuration. (the configuration of the port and path depends on how you configured the local back-end.

Reviewer: Maybe it's better to override those settings in webpack.config.js instead of changing buildConfig.js ?

Author: I know is more clean to override it, but you have anyway copy `buildConfig.js` configuration outside, and setting all missing buildConfig parameters, transforming webpack.config.js in something like this, that is very discouraging: 
```javascript
const themeEntries = require('../themes.js').themeEntries;
const extractThemesPlugin = require('../themes.js').extractThemesPlugin;

const path = require("path");

module.exports = require('../buildConfig')(
    {
        extensions: path.join(__dirname, "app")
    },
    themeEntries,
    {
        base: path.join(__dirname, "..", ".."),
        dist: path.join(__dirname, "dist"),
        framework: path.join(__dirname, "..", "..", "web", "client"),
        code: path.join(__dirname)
    },
    extractThemesPlugin,
    false,
    "/dist/",
    undefined,
    undefined,
    undefined,
    {
        '/rest': {
            target: "https://dev.mapstore.geo-solutions.it/mapstore",
            secure: false,
            headers: {
                host: "dev.mapstore.geo-solutions.it"
            }
        },
        '/pdf': {
            target: "https://dev.mapstore.geo-solutions.it/mapstore",
            secure: false,
            headers: {
                host: "dev.mapstore.geo-solutions.it"
            }
        },
        '/mapstore/pdf': {
            target: "https://dev.mapstore.geo-solutions.it",
            secure: false,
            headers: {
                host: "dev.mapstore.geo-solutions.it"
            }
        },
        '/proxy': {
            target: "https://dev.mapstore.geo-solutions.it/mapstore",
            secure: false,
            headers: {
                host: "dev.mapstore.geo-solutions.it"
            }
        },
        '/docs': {
            target: "http://localhost:8081",
            pathRewrite: { '/docs': '/mapstore/docs' }
        }
    }
);
```
Any suggestion?

I though that placing this configuration in a separate file could help to configure it without taking care of the rest. 
But then this could be a breaking change that makes hard to update depending projects. We planned to refactor `buildConfig.json`, so didn't wanted to do it twice.

Reviewer: I think transforming the buildConfig.js parameters, as we already mentioned other times, into an object would  help, so you can only override what you really need to, but it was only an advice, feel free to ignore it

Author: I want to do it too. This is a breaking change we should add to the list of the things to do when migrate projects, updating the documentation accordingly. 
This issue is need to be backported so can't introduce (or, at least I shouldn't) breaking changes. But I'm totally agree with you.
